### PR TITLE
Add checkpointing and improve flow logging

### PR DESCRIPTION
**Pull request by EthanMarx**

### CODE REVIEW

**Comment by EthanMarx** [comment]
@ravioli1369 once merged, adding `checkpoint` under the model init_args should work with the `test` subcommand

File: projects/train/configs/flow/cbc.yaml [hunk 1]
```diff
@@ -20,6 +20,8 @@ model:
   class_path: train.models.flow.FlowModel
   init_args:
     outdir: ${oc.env:AMPLFI_OUTDIR}
+    # uncomment below when running `trainer.test` to load in a checkpoint
+    # checkpoint:
```
Comment (deepchatterjeeligo) [inline projects/train/configs/flow/cbc.yaml:24]: This looks good. Is it possible to keep something like Null or None value instead of uncommenting?

Comment (EthanMarx) [reply projects/train/configs/flow/cbc.yaml:24]: Ah yeah good call I can make it `null`

Comment (EthanMarx) [reply projects/train/configs/flow/cbc.yaml]: done

**Pull request merged**